### DISCUSSION TITLE
Method call runs two times

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#7922] [1. Help]
Our app has Excel reports. We use laravel-excel. But on my dev-environment it consistently returns 'Failed to open stream: No such file or directory'. This is what happens:

```
calls: [{path: "", method: "export", params: []}]
0 : {path: "", method: "export", params: []}
```

There's only request going back to the app. But when I added a `ray()` call to the `call`-method of `SupportFileDownloads`, I noticed it was called twice. There were two traces shown in Ray.

This explains why I'm getting the 'No such file or directory error' -- `BinaryFileResponse` cleans up after itself. And as there's only one export generated, the file is removed before the second call finishes. When I chain `->deleteFileAfterSend(false)` everything works but it's not the desired behaviour. What I don't understand is:

- How can this happen?
- Why isn't the first file downloading at all--if I remove the clean-up logic from BinaryFileResponse it's also downloaded once which is weird as it's called two times.

I'm lost. Especially as I'm not seeing this behaviour on production. I pretty sure this isn't a bug in Livewire as I've tested it on a fresh install and couldn't reproduce it, but something is causing this to happen. 

Hopefully someone knows what's causing this. I've tried to debug this but got lost. I think it's related to the `EventBus` as there are two sets of the same middlewares loaded but I'm not 100% sure and don't know enough about the internals.

**Comment by joshhanley** [comment]
@ju5t hmm hard to say if you can't replicate the issue. I would try removing code until all you are left with is a file input and one property and see if it still happens. If you manage to reproduce it though, we can then have a look to try and explain what is going on.

**Reply by ju5t** [reply]
@joshhanley I haven't been able to reproduce it in a test repository yet, but when comparing results I found some weird things.

- In our app there are 4 listeners registered when called through `trigger()` in the `EventBus`.
- In the default installation there are only 2.

Both `SupportFileUploads` and `ComponentHookRegistry` are both registered twice in our app. When I test this on a fresh installation they are registered only once. I'll try to debug this further and I'm hoping we don't magically run into this issue in production at one point.

Edit: after some additional testing it's as if the LivewireServiceProvider is mounted 2 times too. Ehh? This is only happening with Livewire.

**Reply by ju5t** [reply]
@joshhanley I've cloned a fresh copy of our app in a new folder and that seems to solve it. Apparently it's some caching that's not being removed even though I cleared it all several times.

**Reply by ju5t** [reply]
@joshhanley turns out cloning a fresh copy only works for a while until the issue comes back. I spent the better part of the day debugging this and I'm not one step closer than I was before.

It's definitely a v3 issue (in combination with our app). As soon as I switch our app to a branch with v2 all issues are gone.

Our app is reasonably sized. It's not easy to slim it down but when I tried, the issue was still there. I don't know what to do anymore, or where to look, but I need to solve this somehow as it's difficult to test certain features. I hope you might have some points where to look. It's weird that the `boot`-method on the `LivewireServiceProvider` is called twice for example.

**Reply by lrljoe** [reply]
Are you seeing any browser console errors?

Otherwise, I'd suspect maybe that an event listener somewhere might be misbehaving and dispatching a linked event, so it's ending up running again.  Check what listeners you have running, and the events they respond to
